File: Task/Continued-fraction/Visual-Basic-.NET/continued-fraction.vb
Module Module1
    Function Calc(f As Func(Of Integer, Integer()), n As Integer) As Double
        Dim temp = 0.0
        For ni = n To 1 Step -1
            Dim p = f(ni)
            temp = p(1) / (p(0) + temp)
        Next
        Return f(0)(0) + temp
    End Function

    Sub Main()
        Dim fList = {
            Function(n As Integer) New Integer() {If(n > 0, 2, 1), 1},
            Function(n As Integer) New Integer() {If(n > 0, n, 2), If(n > 1, n - 1, 1)},
            Function(n As Integer) New Integer() {If(n > 0, 6, 3), Math.Pow(2 * n - 1, 2)}
            }

        For Each f In fList
            Console.WriteLine(Calc(f, 200))
        Next
    End Sub

End Module

File: Task/Continued-fraction/PascalABC.NET/continued-fraction.pas
##
function calc(fun: integer-> (integer, integer); n: integer): decimal;
begin
  var temp := decimal(0.0);
  for var ni := n to 1 step -1 do
  begin
    var (a, b) := fun(ni);
    temp := b / (a + temp);
  end;
  result := fun(0)[0] + temp;
end;

function fsqrt2(n: integer) := (if n > 0 then 2 else 1, 1);

function fnapier(n: integer) := (if n > 0 then n else 2, if n > 1 then n - 1 else 1);

function fpi(n: integer) := (if n > 0 then 6 else 3, (2 * n - 1) * (2 * n - 1));

println(calc(fsqrt2, 200));
println(calc(fnapier, 200));
println(calc(fpi, 10000));
